app/src/main/java/com/nononsenseapps/feeder/model/FeedParser.kt

```diff
@@ -204,6 +235,132 @@ class FeedParser(override val di: DI) : DIAware {
      */
     private suspend fun curl(url: URL) = client.curl(url)
 
+    private suspend fun parseNostrUri(adaptedNostrUri: String): Nip19Profile {
+        val nostrUri = adaptedNostrUri.removePrefix("https://njump.me/")
+        if (nostrUri.contains("@")) { // It means it is a Nip05 address
+            val rawString = nostrUri.removePrefix("nostr:")
+            val parsedNip5 = getNip05Profile(rawString)
+            val (pubkey, relays) = parsedNip5.publicKey() to parsedNip5.relays()
+            return Nip19Profile(pubkey, relays)
+        } else {
+            val parsedProfile = Nip21.parse(nostrUri).asEnum()
+            when (parsedProfile) {
+                is Nip21Enum.Pubkey -> return Nip19Profile(parsedProfile.publicKey)
+                is Nip21Enum.Profile -> return Nip19Profile(parsedProfile.profile.publicKey(), parsedProfile.profile.relays())
+                else -> throw Throwable(message = "Could not find the user's info: $nostrUri")
```

Comment:
You should never throw a `Throwable` directly. Throw a real exception instead.

Reply:
See `FetchError` for example

Reply:
> You should never throw a `Throwable` directly. Throw a real exception instead.

I could create a particular 'NostrUriParserException' for this.